pods/status.go

```diff
@@ -14,13 +14,18 @@ import (
 	log "k8s.io/klog/v2"
 )
 
+// MaxPullTime is how long we will give a pod to start after seeing
+// ErrImagePull.
+var MaxPullTime = time.Minute * 15
```

Comment:
15 minutes seems generous, but I suppose it's better than the current situation where it waits indefinitely!

Reply:
I made it public so it can be turned into a flag :-)

We have tests that fail if they take more than an hour.  I decide 15 minutes was sufficient to a) it really probably will never succeed and b) short enough that it won't accidentally timeout a test.